### PR TITLE
[GlobalISel] Use LocationSize in GISelAddressing. NFC

### PR DESCRIPTION
**Pull request by davemgreen**
This is similar to #83017 but for the areas in GlobalISel's LoadStoreOpt, and should help simplify #70452 a little. It will likely change a little again once the sizes can be scalable.

### CODE REVIEW

File: llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp [hunk 1]
```diff
@@ -117,8 +117,12 @@ bool GISelAddressing::aliasIsKnownForLoadStore(const MachineInstr &MI1,
...
     return false;
 
-  int64_t Size1 = LdSt1->getMemSize();
-  int64_t Size2 = LdSt2->getMemSize();
+  LocationSize Size1 = LdSt1->getMemSize() != MemoryLocation::UnknownSize
+                           ? LdSt1->getMemSize()
+                           : LocationSize::beforeOrAfterPointer();
+  LocationSize Size2 = LdSt2->getMemSize() != MemoryLocation::UnknownSize
+                           ? LdSt2->getMemSize()
+                           : LocationSize::beforeOrAfterPointer();
```
Comment (aemerson) [inline llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp:125]: How about we add helpers to `GLoadStore` like `getMemLocationSize()`. I find having to compare against this awkwardly named `beforeOrAfterPointer()` thing cumbersome.

**Review by aemerson — APPROVED** [review]
LGTM with a suggestion.

File: llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp (hunk 1 above)
Comment (davemgreen) [reply llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp:125]: Thanks - The plan is for getMemSize() to become a LocationSize in #70452, so this should be cleaned up to a simpler `LocationSize Size1 = LdSt1->getMemSize();`. It just has to be a bit awkward for the moment, and eventually getMemSize will never be MemoryLocation::UnknownSize.

Comment (aemerson) [reply llvm/lib/CodeGen/GlobalISel/LoadStoreOpt.cpp:125]: Ok go ahead then.

**Pull request merged**